src/controllers/teams.controller/deleteTeamDocument/index.js

```diff
@@ -0,0 +1,61 @@
+const { validationResult } = require('express-validator');
+const catchAsync = require('../../../utils/catchAsync');
+const { query: Hasura } = require('../../../utils/hasura');
+const { getTeamDocumentQuery, checkIfMemberQuery, deleteTeamDocumentQuery } = require('./queries/queries');
+const deleteDocument = require('../../../utils/deleteDocument');
+
+const deleteTeamDocument = catchAsync(async (req, res) => {
+  const sanitizerErrors = validationResult(req);
+  if (!sanitizerErrors.isEmpty()) {
+    return res.status(400).json({
+      success: false,
+      errors: sanitizerErrors.array(),
+    });
+  }
+
+  const { cognito_sub: cognitoSub, team_id: teamId, documentId } = req.body;
+
+  // Check if the user is a member of the team
+  const memberCheckResult = await Hasura(checkIfMemberQuery, {
+    cognitoSub,
+    teamId,
+  });
+
+  if (memberCheckResult.result.data.team_members.length === 0) {
+    return res.status(403).json({
+      success: false,
+      errorCode: 'FORBIDDEN',
+      errorMessage: 'You are not authorized to perform this action',
+    });
+  }
+
+  const documentQueryResult = await Hasura(getTeamDocumentQuery, {
```

Comment:
Good naming choice. Can we alter it to match the query name, so its even easier to understand which query's response it contains? like `getTeamDocumentResult`?